### PR TITLE
issue#254_add_PGID-UID_to_Rockon_doc

### LINKED ISSUE #254: Add PGID/UID reference to General Rockon section
Since quite a few containers (with and without write-up) have a requirement to enter PGID or UID (especially if they are based on linuxserver.io's templates), should we add a section in the general Rockon section (overview.rst), that either references the UID/PGID section in the rockstor install, or give a couple of ways to obtain that information (the primary one being for the less technical audience and show the WebUI option.
In the Plex write-up we also had a section to create a new Rockon specific user, so that's a question from me whether that should also be noted in this general context (either recap or reference Rockstor section, or reference Plex section on new user creation, etc.).

### PR DESCRIPTION
**Pull request by Hooverdan96**
Fixes #254 

### This pull request's proposal
Adding small subsection mentioning UID and GID usage and where to find them during Rock-on configuration.

### Checklist
- [X] With the proposed changes no Sphinx errors or warnings are generated.
- [ ] I have added my name to the AUTHORS file, if required (descending alphabetical order).

Log output:
Running Sphinx v3.3.1
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 1 source files that are out of date
updating environment: 0 added, 1 changed, 0 removed
reading sources... [100%] docker-based-rock-ons/overview                                                                                                                                                                     
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] index                                                                                                                                                                                               
generating indices... genindex done
writing additional pages... search done
copying images... [100%] docker-based-rock-ons/addlabels_labels_selection.png                                                                                                                                                
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.

### CODE REVIEW

File: docker-based-rock-ons/overview.rst [hunk 1]
```diff
@@ -76,6 +76,19 @@ If no Rock-ons are showing on the **All** tab then click the **Update** button
 to refresh the list of available Rock-ons. To install a listed Rock-on use
 its **Install** button on the Rock-ons WebUI page.
 
+.. _user_group:
```
Comment (FroggyFlox) [inline docker-based-rock-ons/overview.rst:79]: I would prefer further specification to differentiate it from our Users/Groups dedicated page:
```suggestion
.. _rockons_user_group:
```

File: docker-based-rock-ons/overview.rst [hunk 2]
```diff
@@ -76,6 +76,19 @@ If no Rock-ons are showing on the **All** tab then click the **Update** button
...
+
+UID and GID usage in Rock-ons
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+During the configuration of Rock-ons, quite a few require the specification of a User ID (UID) and/or a User Group ID (GID). Aside from the various options using the command line to determine existing or creating new UIDs or GIDs, the simplest way is to take advantage of the User and/or Group management screen. Navigate to System --> Users (or Groups if necessary).
```
Comment (FroggyFlox) [inline docker-based-rock-ons/overview.rst:84]: Minor style edits (rst style recommends line length of 79 characters max):
```suggestion
During the configuration of Rock-ons, quite a few require the specification of a
User ID (UID) and/or a Group ID (GID). Aside from the various options using the
command line to determine existing UIDs and GIDs, the simplest way is to take
advantage of the User and/or Group management screens. Navigate to System
--> Users (or Groups if necessary).
```

File: docker-based-rock-ons/overview.rst [hunk 3]
```diff
@@ -76,6 +76,19 @@ If no Rock-ons are showing on the **All** tab then click the **Update** button
...
+
+.. image:: users_page.png
+   :scale: 30%
+   :align: center
+   
+The two highlighted columns show the UID and GID. For example, the user plex had the UID 1001, as well as the GID 1001, which could then be used to set up the Plex Rock-on.
```
Comment (FroggyFlox) [inline docker-based-rock-ons/overview.rst:90]: Minor style edits:
```suggestion
The two highlighted columns show the UID and GID. In the example above, the
user :code:`plex` has the UID 1001, as well as the GID 1001, which could then
be used to set up the Plex rock-on.
```

**Review by FroggyFlox — CHANGES_REQUESTED** [review]
@Hooverdan96 , nicely done!

Thanks a lot for improving our documentation on this, this was indeed an important point to detail so it's really helpful to have it in our documentation!

I only have a few minor style edits, mostly related to style recommendations for RST files:
- max character lengths: 79
- no trailing space at the end of line

I'm looking into finding a way to automate this but for the moment we need to do this manually, unfortunately.
Once these edits are in, I'll test them once more before merging your work.

Thanks again!

**Comment by Hooverdan96** [comment]
@FroggyFlox - thanks for the review. All of the above suggestions are sensible, and I have committed them. I obviously didn't read the style guide, so thanks for pointing out the 79 character max limit. I was looking for a UI based Sphinx editor but had only found an online one at this time. I will remember for next time.

**Review by FroggyFlox — APPROVED** [review]

**Comment by FroggyFlox** [comment]
Thanks for the edits, @Hooverdan96!

Everything looks well in my most recent test so I'm going to merge and publish it now. Thanks again for your continued efforts and contributions!

**Pull request merged**